pinot-core/src/main/java/com/linkedin/pinot/core/segment/store/SegmentDirectoryPaths.java

```diff
@@ -15,60 +15,89 @@
  */
 package com.linkedin.pinot.core.segment.store;
 
-import com.google.common.base.Preconditions;
 import com.linkedin.pinot.core.indexsegment.generator.SegmentVersion;
 import com.linkedin.pinot.core.segment.creator.impl.V1Constants;
 import java.io.File;
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 
 public class SegmentDirectoryPaths {
-  public static final String V3_SUBDIRECTORY_NAME = "v3";
-  private static final Logger LOGGER = LoggerFactory.getLogger(SegmentDirectoryPaths.class);
+  private SegmentDirectoryPaths() {
+  }
 
+  public static final String V3_SUBDIRECTORY_NAME = "v3";
 
-  public static File segmentDirectoryFor(File segmentIndexDirectory, SegmentVersion version) {
-    switch(version) {
+  @Nonnull
+  public static File segmentDirectoryFor(@Nonnull File indexDir, @Nonnull SegmentVersion segmentVersion) {
+    switch (segmentVersion) {
       case v1:
       case v2:
-        return segmentIndexDirectory;
+        return indexDir;
       case v3:
-        return new File(segmentIndexDirectory, V3_SUBDIRECTORY_NAME);
+        return new File(indexDir, V3_SUBDIRECTORY_NAME);
       default:
-        throw new UnsupportedOperationException("Segment path for version: " + version +
-            " and segmentIndexDirectory: " + segmentIndexDirectory + " can not be determined ");
+        throw new UnsupportedOperationException("Unsupported segment version: " + segmentVersion);
     }
   }
 
-  public static boolean isV3Directory(File path) {
+  public static boolean isV3Directory(@Nonnull File path) {
     return path.toString().endsWith(V3_SUBDIRECTORY_NAME);
   }
 
-  public static @Nullable File findMetadataFile(@Nonnull File indexDir) {
+  @Nullable
+  public static File findMetadataFile(@Nonnull File indexDir, @Nonnull SegmentVersion segmentVersion) {
+    return findFormatFile(indexDir, V1Constants.MetadataKeys.METADATA_FILE_NAME, segmentVersion);
+  }
+
+  @Nullable
+  public static File findMetadataFile(@Nonnull File indexDir) {
     return findFormatFile(indexDir, V1Constants.MetadataKeys.METADATA_FILE_NAME);
   }
 
-  /**
-   * Find the creation metadata file in the segment directory path to
-   * handle different segment formats
-   * @param indexDir index directory to look for file
-   * @return creation metadata file or null if the file is not found
-   */
-  public static @Nullable File findCreationMetaFile(@Nonnull File indexDir) {
+  @Nullable
+  public static File findCreationMetaFile(@Nonnull File indexDir, @Nonnull SegmentVersion segmentVersion) {
```

Comment:
Preconditions checks have performance impact, that's the reason we prefer annotations. IDE will mark warnings if any code does not follow the convention.